python/taichi/profiler/kernelmetrics.py

```diff
@@ -0,0 +1,200 @@
+from dataclasses import dataclass
+
+from taichi.core import ti_core as _ti_core
+
+
+@dataclass
+class CuptiMetric:
+    """A data class to add CUPTI metric for :class:`~taichi.lang.KernelProfiler`.
+
+    This data class is designed to add user selected CUPTI metrics.
+    Only available for the CUDA backend now, i.e. you need ``ti.init(kernel_profiler=True, arch=ti.cuda)``.
+    For usage of this class, see examples in func :func:`~taichi.lang.set_kernel_profile_metrics` and :func:`~taichi.lang.collect_kernel_profile_metrics_in_context`.
+    For prerequisites to using CUPTI in Taichi, please visit https://docs.taichi.graphics/docs/lang/articles/misc/profiler#advanced-mode.
```

Comment:
We don't have to repeatedly write this

> For prerequisites to using CUPTI in Taichi, please visit https://docs.taichi.graphics/docs/lang/articles/misc/profiler#advanced-mode.
        For details about CUPTI, please visit https://docs.nvidia.com/cupti/Cupti/index.html.
        For properties of your GPU metrics, build and run the sample in path ``/usr/local/cuda/extras/CUPTI/samples/cupti_metric_properties``.

Just put it somewhere in the docs?